src/bindings/scl.rs

```diff
@@ -0,0 +1,261 @@
+                s += &k;
+                s += "=";
+                s += &to_string(v);
+                s += ",";
+            }
+            s += "}";
+            s
+        }
+    }
+}
+
+pub fn init(lua: &rlua::Lua) -> Result<(), rlua::Error> {
+    // Decode string to a table
+    let module = lua.create_table()?;
+    module.set(
+        "to_table",
+        lua.create_function(|lua, text: String| {
+            let dict: scl::Dict = scl::parse_str(&text).unwrap();
+            Ok(to_rlua_value(lua, scl::Value::Dict(dict)))
+        })?,
+    )?;
+
+    // Encode table to a string
+    module.set(
+        "from_table",
+        lua.create_function(|lua, table: rlua::Table| {
+            let val = to_scl_value(lua, rlua::Value::Table(table));
+            if let scl::Value::Dict(d) = val {
+                Ok(scl_decode(d))
+            } else {
+                unreachable!();
+            }
+        })?,
+    )?;
+
+    lua.globals().set("scl", module)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_decode() {
+        static TEXT: &str = r##"
+owner = {
+  name = "Vincent Prouillet",
+  dob = 1979-05-27, # first-class date type
+}
+
+database = {
+  ports = [ 8001, 8001, 8002 ],
+  connection_max = 5000,
+  enabled = true,
+}
+
+servers = {
+  max_upload_size = 10MB, # first class byte size
+  alpha = {
+    ip = "10.0.0.1",
+    dc = "eqdc10",
+  },
+}
+
+clients = {
+  data = [
+    ["gamma"],
+    [1],
+  ],
+}
+        "##;
+        let lua = rlua::Lua::new();
+        super::init(&lua).unwrap();
+        lua.globals().set("TEXT", TEXT).unwrap();
+        lua.exec::<_, rlua::Value>(
+            r##"
+        local x = scl.to_table(TEXT)
+        assert(x.owner.name == "Vincent Prouillet")
+        assert(x.owner.dob.year == 1979)
+        assert(x.owner.dob.month == 5)
+        assert(x.owner.dob.day == 27)
+        assert(#x.database.ports == 3)
+        assert(x.database.ports[1] == 8001)
+        assert(x.database.ports[2] == 8001)
+        assert(x.database.ports[3] == 8002)
+        assert(x.database.connection_max == 5000)
+        assert(x.database.enabled == true)
+        assert(x.servers.max_upload_size == 10000000)
+        assert(x.servers.alpha.ip == "10.0.0.1")
+        assert(x.servers.alpha.dc == "eqdc10")
+        assert(x.clients.data[1][1] == "gamma")
+        assert(x.clients.data[2][1] == 1)
+        "##,
+            None,
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn test_encode() {
+        let lua = rlua::Lua::new();
+        super::init(&lua).unwrap();
+        lua.exec::<_, rlua::Value>(
+            r##"
+        local x = {
+        x=0,
+        y=true,
+        z="a\nb",
+        a=1.2,
+        b={c=1},
+        d={day=1,month=2,year=2018},
+        e={2, 3, 4},
+        }
+        local s = scl.from_table(x)
+        print(s)
+        local y = scl.to_table(s)
+        -- don't mess with floating point number
+        -- assert(y.a == 1.2)
```

Comment:
What's up with this? Why doesn't it work?

Reply:
Nothing wrong with it, I just don't want to compare floating point numbers.

Reply:
why not use 1.5? that's perfectly representable

Reply:
I think you can push to my branch, so change what you want about it.